## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#b38cf153-07f9-43cd-80e4-ff11a5e54bbd) |
| mint-gateway-s3.sh | :x: [more...](#133f9b00-a271-4800-b81e-5863bf9a53ba) |
| mint-gateway-azure.sh | :x: [more...](#43de5cc1-5f7e-4268-b4b9-a43147c64522) |
| mint-erasure.sh | :x: [more...](#37357699-ec69-4845-82f9-dbb5cc173361) |
| mint-dist-erasure.sh | :x: [more...](#d98d5317-79e5-4099-8896-edbf25df8d52) |
| mint-zoned.sh | :x: [more...](#0499819b-fa9c-4dec-82d8-fbe74c737217) |
<a name="37357699-ec69-4845-82f9-dbb5cc173361"></a>
### 11280-dc0cf7e/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:30272
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 9f7e39720bf3:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... FAILED in 42 seconds
PHP Warning:  strtolower() expects parameter 1 to be string, array given in /mint/run/core/aws-sdk-php/quick-tests.php on line 189
PHP Warning:  strtolower() expects parameter 1 to be string, array given in /mint/run/core/aws-sdk-php/quick-tests.php on line 189
PHP Warning:  strtolower() expects parameter 1 to be string, array given in /mint/run/core/aws-sdk-php/quick-tests.php on line 189
PHP Warning:  strtolower() expects parameter 1 to be string, array given in /mint/run/core/aws-sdk-php/quick-tests.php on line 189
PHP Fatal error:  Uncaught exception 'Aws\S3\Exception\S3Exception' with message 'Error executing "DeleteObject" on "http://minio-dev8.minio.io:30272/aws-sdk-php-60011d28e1c28/obj1"; AWS HTTP error: Client error: `DELETE http://minio-dev8.minio.io:30272/aws-sdk-php-60011d28e1c28/obj1` resulted in a `404 Not Found` response:
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>NoSuchBucket</Code><Message>The specified bucket does not exist</Mes (truncated...)
 NoSuchBucket (client): The specified bucket does not exist - <?xml version="1.0" encoding="UTF-8"?>
<Error><Code>NoSuchBucket</Code><Message>The specified bucket does not exist</Message><Key>obj1</Key><BucketName>aws-sdk-php-60011d28e1c28</BucketName><Resource>/aws-sdk-php-60011d28e1c28/obj1</Resource><RequestId>165A4E2E857C62E5</RequestId><HostId>9376833f-d305-4f22-9e6e-bee2cebe61b9</HostId></Error>'

GuzzleHttp\Exception\ClientException: Client error: `DELETE http://minio-dev8.minio.io:30272/aws-sdk-php-60011d28e1c28/obj1` resulted in a `404 Not Found` response:
<?xml ve in /mint/run/core/aws-sdk-php/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php on line 195
(3/15) Running aws-sdk-ruby tests ... done in 2 seconds
(4/15) Running awscli tests ... FAILED in 57 seconds
{"name": "awscli", "duration": 15830, "function": "aws --endpoint-url http://minio-dev8.minio.io:30272 s3api put-object --body /mint/data/datafile-10-MB --bucket awscli-mint-test-bucket-28725 --key datafile-10-MB\n", "status": "FAIL", "error": "Could not connect to the endpoint URL: "http://minio-dev8.minio.io:30272/awscli-mint-test-bucket-28725/datafile-10-MB""}
(4/15) Running healthcheck tests ... done in 0 seconds
(5/15) Running mc tests ... done in 44 seconds
(6/15) Running minio-dotnet tests ... done in 58 seconds
(7/15) Running minio-go tests ... FAILED in 51 seconds
{
  "args": {
    "bucketName": "minio-go-test-yovm5h6vsex4kusm",
    "fileName": "",
    "objectContentType": "testapplication/octet-stream",
    "objectName": "minio-go-test-yovm5h6vsex4kusmFPutObject-standard",
    "opts": ""
  },
  "duration": 11758,
  "error": "Put \"http://minio-dev8.minio.io:30272/minio-go-test-yovm5h6vsex4kusm/minio-go-test-yovm5h6vsex4kusmFPutObject-standard?partNumber=2&uploadId=f5e0127c-d393-4cee-88eb-2f4b3ce253ba\": dial tcp 72.28.97.61:30272: connect: connection refused",
  "function": "FPutObject(bucketName, objectName, fileName, opts)",
  "message": "FPutObject failed",
  "name": "minio-go: testFPutObjectMultipart",
  "status": "FAIL"
}
(7/15) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 103,
  "status": "FAIL",
  "error": "java.net.ConnectException: Failed to connect to minio-dev8.minio.io/72.28.97.61:30272 >>> [okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:297), okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207), okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226), okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106), okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74), okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall.execute(RealCall.kt:154), io.minio.MinioClient.execute(MinioClient.java:647), io.minio.MinioClient.makeBucket(MinioClient.java:2276), FunctionalTest.testBucketApi(FunctionalTest.java:446), FunctionalTest.testBucketApiCases(FunctionalTest.java:471), FunctionalTest.makeBucket(FunctionalTest.java:502), FunctionalTest.runBucketTests(FunctionalTest.java:3708), FunctionalTest.runTests(FunctionalTest.java:3782), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(7/15) Running minio-js tests ... FAILED in 2 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 1033,
  "status": "FAIL",
  "error": "Error: connect ECONNREFUSED 72.28.97.61:30272 at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:16)"
}
(7/15) Running minio-py tests ... FAILED in 8 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-dd86529e-8185-4662-a247-2cecb19486e9",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='minio-dev8.minio.io', port=30272): Max retries exceeded with url: /minio-py-test-dd86529e-8185-4662-a247-2cecb19486e9 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fb04fb2f588>: Failed to establish a new connection: [Errno 111] Connection refused',))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 170, in _new_conn\n    (self._dns_host, self.port), self.timeout, **extra_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py\", line 96, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py\", line 86, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 706, in urlopen\n    chunked=chunked,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 394, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 234, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.6/http/client.py\", line 1281, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1327, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1276, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1042, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.6/http/client.py\", line 980, in send\n    self.connect()\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 200, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 182, in _new_conn\n    self, \"Failed to establish a new connection: %s\" % e\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7fb04fb2f588>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 169, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 624, in make_bucket\n    headers=headers,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 278, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 756, in urlopen\n    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 573, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-dev8.minio.io', port=30272): Max retries exceeded with url: /minio-py-test-dd86529e-8185-4662-a247-2cecb19486e9 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fb04fb2f588>: Failed to establish a new connection: [Errno 111] Connection refused',))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 8129
}
(7/15) Running s3cmd tests ... FAILED in 1 seconds
{
  "name": "s3cmd",
  "duration": "308",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  following lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\nInvoked as: /usr/local/bin/s3cmd --config /tmp/.s3cmd-12692/s3cfg mb s3://s3cmd-test-bucket-3351\nProblem: <class 'ConnectionRefusedError: [Errno 111] Connection refused\nS3cmd:   2.1.0\npython:   3.6.9 (default, Oct  8 2020, 12:12:24) \n[GCC 8.4.0]\nenvironment LANG=C.UTF-8\n\nTraceback (most recent call last):\n  File \"/usr/local/bin/s3cmd\", line 3121, in <module>\n    rc = main()\n  File \"/usr/local/bin/s3cmd\", line 3030, in main\n    rc = cmd_func(args)\n  File \"/usr/local/bin/s3cmd\", line 251, in cmd_bucket_create\n    response = s3.bucket_create(uri.bucket(), cfg.bucket_location)\n  File \"/usr/local/lib/python3.6/dist-packages/S3/S3.py\", line 408, in bucket_create\n    response = self.send_request(request)\n  File \"/usr/local/lib/python3.6/dist-packages/S3/S3.py\", line 1281, in send_request\n    conn = ConnMan.get(self.get_hostname(resource['bucket']))\n  File \"/usr/local/lib/python3.6/dist-packages/S3/ConnMan.py\", line 251, in get\n    conn.c.connect()\n  File \"/usr/lib/python3.6/http/client.py\", line 952, in connect\n    (self.host,self.port), self.timeout, self.source_address)\n  File \"/usr/lib/python3.6/socket.py\", line 724, in create_connection\n    raise err\n  File \"/usr/lib/python3.6/socket.py\", line 713, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  above lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
}
(7/15) Running s3select tests ... FAILED in 6 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-79b5e69e-2aff-45e6-8561-fb61790a6c62"
  },
  "duration": 6072,
  "message": "HTTPConnectionPool(host='minio-dev8.minio.io', port=30272): Max retries exceeded with url: /s3select-test-79b5e69e-2aff-45e6-8561-fb61790a6c62 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fd644790f60>: Failed to establish a new connection: [Errno 111] Connection refused',))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 170, in _new_conn\n    (self._dns_host, self.port), self.timeout, **extra_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py\", line 96, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py\", line 86, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 706, in urlopen\n    chunked=chunked,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 394, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 234, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.6/http/client.py\", line 1281, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1327, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1276, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1042, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.6/http/client.py\", line 980, in send\n    self.connect()\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 200, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 182, in _new_conn\n    self, \"Failed to establish a new connection: %s\" % e\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7fd644790f60>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 624, in make_bucket\n    headers=headers,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 278, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 756, in urlopen\n    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 573, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-dev8.minio.io', port=30272): Max retries exceeded with url: /s3select-test-79b5e69e-2aff-45e6-8561-fb61790a6c62 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fd644790f60>: Failed to establish a new connection: [Errno 111] Connection refused',))\n",
  "status": "FAIL"
}
(7/15) Running security tests ... done in 0 seconds

Executed 7 out of 15 tests successfully.
```
<a name="133f9b00-a271-4800-b81e-5863bf9a53ba"></a>
### 11280-dc0cf7e/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31528
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp c0a008b1c177:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... FAILED in 43 seconds
PHP Warning:  strtolower() expects parameter 1 to be string, array given in /mint/run/core/aws-sdk-php/quick-tests.php on line 189
PHP Warning:  strtolower() expects parameter 1 to be string, array given in /mint/run/core/aws-sdk-php/quick-tests.php on line 189
PHP Warning:  strtolower() expects parameter 1 to be string, array given in /mint/run/core/aws-sdk-php/quick-tests.php on line 189
PHP Warning:  strtolower() expects parameter 1 to be string, array given in /mint/run/core/aws-sdk-php/quick-tests.php on line 189
PHP Fatal error:  Uncaught exception 'Aws\S3\Exception\S3Exception' with message 'Error executing "DeleteObject" on "http://minio-c3.minio.io:31528/aws-sdk-php-60011d2a14398/obj1"; AWS HTTP error: Client error: `DELETE http://minio-c3.minio.io:31528/aws-sdk-php-60011d2a14398/obj1` resulted in a `404 Not Found` response:
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>NoSuchBucket</Code><Message>The specified bucket does not exist</Mes (truncated...)
 NoSuchBucket (client): The specified bucket does not exist - <?xml version="1.0" encoding="UTF-8"?>
<Error><Code>NoSuchBucket</Code><Message>The specified bucket does not exist</Message><Key>obj1</Key><BucketName>aws-sdk-php-60011d2a14398</BucketName><Resource>/aws-sdk-php-60011d2a14398/obj1</Resource><RequestId>165A4E2EE3F1515E</RequestId><HostId>1e02fbd1-d234-418e-b85c-78a10fb6d755</HostId></Error>'

GuzzleHttp\Exception\ClientException: Client error: `DELETE http://minio-c3.minio.io:31528/aws-sdk-php-60011d2a14398/obj1` resulted in a `404 Not Found` response:
<?xml version= in /mint/run/core/aws-sdk-php/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php on line 195
(3/15) Running aws-sdk-ruby tests ... done in 2 seconds
(4/15) Running awscli tests ... FAILED in 57 seconds
{
  "name": "awscli",
  "duration": 18238,
  "function": "delete_bucket\n",
  "status": "FAIL",
  "error": "delete: s3://awscli-mint-test-bucket-3040/datafile-1-kBremove_bucket failed: s3://awscli-mint-test-bucket-3040 An error occurred (NoSuchBucket) when calling the DeleteBucket operation: The specified bucket does not exist"
}
(4/15) Running healthcheck tests ... done in 0 seconds
(5/15) Running mc tests ... done in 41 seconds
(6/15) Running minio-dotnet tests ... done in 41 seconds
(7/15) Running minio-go tests ... FAILED in 1 minutes and 8 seconds
{
  "args": {
    "destination": "",
    "sourceList": ""
  },
  "duration": 1911,
  "error": "Put \"http://minio-c3.minio.io:31528/minio-go-test-cipdj539p5kfkgy5/srcObject\": dial tcp 72.28.97.59:31528: connect: connection refused",
  "function": "ComposeObject(destination, sourceList)",
  "message": "PutObject failed",
  "name": "minio-go: testComposeMultipleSources",
  "status": "FAIL"
}
(7/15) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 76,
  "status": "FAIL",
  "error": "java.net.ConnectException: Failed to connect to minio-c3.minio.io/72.28.97.59:31528 >>> [okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:297), okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207), okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226), okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106), okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74), okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall.execute(RealCall.kt:154), io.minio.MinioClient.execute(MinioClient.java:647), io.minio.MinioClient.makeBucket(MinioClient.java:2276), FunctionalTest.testBucketApi(FunctionalTest.java:446), FunctionalTest.testBucketApiCases(FunctionalTest.java:471), FunctionalTest.makeBucket(FunctionalTest.java:502), FunctionalTest.runBucketTests(FunctionalTest.java:3708), FunctionalTest.runTests(FunctionalTest.java:3782), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(7/15) Running minio-js tests ... FAILED in 2 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 1044,
  "status": "FAIL",
  "error": "Error: connect ECONNREFUSED 72.28.97.59:31528 at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:16)"
}
(7/15) Running minio-py tests ... FAILED in 8 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-ab13b89e-50a2-426b-be69-af464ae260af",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='minio-c3.minio.io', port=31528): Max retries exceeded with url: /minio-py-test-ab13b89e-50a2-426b-be69-af464ae260af (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f03c3b35550>: Failed to establish a new connection: [Errno 111] Connection refused',))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 170, in _new_conn\n    (self._dns_host, self.port), self.timeout, **extra_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py\", line 96, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py\", line 86, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 706, in urlopen\n    chunked=chunked,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 394, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 234, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.6/http/client.py\", line 1281, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1327, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1276, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1042, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.6/http/client.py\", line 980, in send\n    self.connect()\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 200, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 182, in _new_conn\n    self, \"Failed to establish a new connection: %s\" % e\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f03c3b35550>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 169, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 624, in make_bucket\n    headers=headers,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 278, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 756, in urlopen\n    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 573, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-c3.minio.io', port=31528): Max retries exceeded with url: /minio-py-test-ab13b89e-50a2-426b-be69-af464ae260af (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f03c3b35550>: Failed to establish a new connection: [Errno 111] Connection refused',))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 8177
}
(7/15) Running s3cmd tests ... FAILED in 1 seconds
{
  "name": "s3cmd",
  "duration": "320",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  following lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\nInvoked as: /usr/local/bin/s3cmd --config /tmp/.s3cmd-27064/s3cfg mb s3://s3cmd-test-bucket-14749\nProblem: <class 'ConnectionRefusedError: [Errno 111] Connection refused\nS3cmd:   2.1.0\npython:   3.6.9 (default, Oct  8 2020, 12:12:24) \n[GCC 8.4.0]\nenvironment LANG=C.UTF-8\n\nTraceback (most recent call last):\n  File \"/usr/local/bin/s3cmd\", line 3121, in <module>\n    rc = main()\n  File \"/usr/local/bin/s3cmd\", line 3030, in main\n    rc = cmd_func(args)\n  File \"/usr/local/bin/s3cmd\", line 251, in cmd_bucket_create\n    response = s3.bucket_create(uri.bucket(), cfg.bucket_location)\n  File \"/usr/local/lib/python3.6/dist-packages/S3/S3.py\", line 408, in bucket_create\n    response = self.send_request(request)\n  File \"/usr/local/lib/python3.6/dist-packages/S3/S3.py\", line 1281, in send_request\n    conn = ConnMan.get(self.get_hostname(resource['bucket']))\n  File \"/usr/local/lib/python3.6/dist-packages/S3/ConnMan.py\", line 251, in get\n    conn.c.connect()\n  File \"/usr/lib/python3.6/http/client.py\", line 952, in connect\n    (self.host,self.port), self.timeout, self.source_address)\n  File \"/usr/lib/python3.6/socket.py\", line 724, in create_connection\n    raise err\n  File \"/usr/lib/python3.6/socket.py\", line 713, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  above lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
}
(7/15) Running s3select tests ... FAILED in 6 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-6a2374ec-f719-40f0-bf07-925c0ce77fac"
  },
  "duration": 6070,
  "message": "HTTPConnectionPool(host='minio-c3.minio.io', port=31528): Max retries exceeded with url: /s3select-test-6a2374ec-f719-40f0-bf07-925c0ce77fac (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f113f8eef98>: Failed to establish a new connection: [Errno 111] Connection refused',))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 170, in _new_conn\n    (self._dns_host, self.port), self.timeout, **extra_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py\", line 96, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py\", line 86, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 706, in urlopen\n    chunked=chunked,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 394, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 234, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.6/http/client.py\", line 1281, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1327, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1276, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1042, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.6/http/client.py\", line 980, in send\n    self.connect()\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 200, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 182, in _new_conn\n    self, \"Failed to establish a new connection: %s\" % e\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f113f8eef98>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 624, in make_bucket\n    headers=headers,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 278, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 756, in urlopen\n    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 573, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-c3.minio.io', port=31528): Max retries exceeded with url: /s3select-test-6a2374ec-f719-40f0-bf07-925c0ce77fac (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f113f8eef98>: Failed to establish a new connection: [Errno 111] Connection refused',))\n",
  "status": "FAIL"
}
(7/15) Running security tests ... done in 0 seconds

Executed 7 out of 15 tests successfully.
```
<a name="0499819b-fa9c-4dec-82d8-fbe74c737217"></a>
### 11280-dc0cf7e/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:30467
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 2dd3c372ebd6:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... FAILED in 43 seconds
{
  "name": "aws-sdk-php",
  "function": "getBucketPolicy ( array $params = [] )",
  "args": {
    "Bucket": "aws-sdk-php-60011d2a876e9"
  },
  "duration": "40247",
  "status": "FAIL",
  "error": "Error executing \"CreateBucket\" on \"http://minio-dev8.minio.io:30467/aws-sdk-php-60011d2a876e9\"; AWS HTTP error: Server error: `PUT http://minio-dev8.minio.io:30467/aws-sdk-php-60011d2a876e9` resulted in a `503 Service Unavailable` response:\n<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>SlowDown</Code><Message>Please reduce your request</Message><BucketN (truncated...)\n SlowDown (server): Please reduce your request - <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>SlowDown</Code><Message>Please reduce your request</Message><BucketName>aws-sdk-php-60011d2a876e9</BucketName><Resource>/aws-sdk-php-60011d2a876e9</Resource><RequestId>165A4E2F4534E76C</RequestId><HostId>bb959b03-7ef0-4e7a-917c-64e5b8ccb2e2</HostId></Error>"
}
(3/15) Running aws-sdk-ruby tests ... done in 7 seconds
(4/15) Running awscli tests ... FAILED in 1 minutes and 14 seconds
{
  "name": "awscli",
  "duration": 23903,
  "function": "aws --endpoint-url http://minio-dev8.minio.io:30467 s3 rb s3://awscli-mint-test-bucket-24794/\n",
  "status": "FAIL",
  "error": "remove_bucket failed: s3://awscli-mint-test-bucket-24794/ An error occurred (SlowDown) when calling the DeleteBucket operation (reached max retries: 4): Please reduce your request"
}
(4/15) Running healthcheck tests ... done in 1 seconds
(5/15) Running mc tests ... FAILED in 12 seconds
{
  "name": "mc",
  "duration": "1736",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-30728 --quiet --no-color mb myminio/mc-test-bucket-9607  >>> \nmc: <ERROR> Unable to make bucket `myminio/mc-test-bucket-9607`. Your previous request to create the named bucket succeeded and you already own it."
}
(5/15) Running minio-dotnet tests ... done in 45 seconds
(6/15) Running minio-go tests ... FAILED in 2 minutes and 44 seconds
{
  "args": {},
  "duration": 1491,
  "error": "unexpected EOF",
  "function": "CopyObjectPart(destination, source)",
  "message": "Read buffer failed",
  "name": "minio-go: testSSES3EncryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(6/15) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 75,
  "status": "FAIL",
  "error": "java.net.ConnectException: Failed to connect to minio-dev8.minio.io/72.28.97.61:30467 >>> [okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:297), okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207), okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226), okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106), okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74), okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall.execute(RealCall.kt:154), io.minio.MinioClient.execute(MinioClient.java:647), io.minio.MinioClient.makeBucket(MinioClient.java:2276), FunctionalTest.testBucketApi(FunctionalTest.java:446), FunctionalTest.testBucketApiCases(FunctionalTest.java:471), FunctionalTest.makeBucket(FunctionalTest.java:502), FunctionalTest.runBucketTests(FunctionalTest.java:3708), FunctionalTest.runTests(FunctionalTest.java:3782), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(6/15) Running minio-js tests ... FAILED in 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 15,
  "status": "FAIL",
  "error": "S3Error: Please reduce your request at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:79:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:684:10) at endReadableNT (internal/streams/readable.js:1327:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(6/15) Running minio-py tests ... FAILED in 2 minutes and 0 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-795ee345-956f-4a25-990f-02f88a6e42bf",
    "location": "default value ('us-east-1')"
  },
  "message": "S3 operation failed; code: BucketAlreadyOwnedByYou, message: Your previous request to create the named bucket succeeded and you already own it., resource: /minio-py-test-795ee345-956f-4a25-990f-02f88a6e42bf, request_id: 165A4E92338F0BE6, host_id: bb959b03-7ef0-4e7a-917c-64e5b8ccb2e2, bucket_name: minio-py-test-795ee345-956f-4a25-990f-02f88a6e42bf",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 169, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 624, in make_bucket\n    headers=headers,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 385, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketAlreadyOwnedByYou, message: Your previous request to create the named bucket succeeded and you already own it., resource: /minio-py-test-795ee345-956f-4a25-990f-02f88a6e42bf, request_id: 165A4E92338F0BE6, host_id: bb959b03-7ef0-4e7a-917c-64e5b8ccb2e2, bucket_name: minio-py-test-795ee345-956f-4a25-990f-02f88a6e42bf\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 120137
}
(6/15) Running s3cmd tests ... done in 22 seconds
(7/15) Running s3select tests ... done in 21 seconds
(8/15) Running security tests ... done in 0 seconds

Executed 8 out of 15 tests successfully.
```
<a name="d98d5317-79e5-4099-8896-edbf25df8d52"></a>
### 11280-dc0cf7e/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:30612
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e609aa93d39c:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 2 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 46 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 2 minutes and 14 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 50 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 1 seconds
(9/15) Running minio-go tests ... done in 2 minutes and 9 seconds
(10/15) Running minio-java tests ... done in 1 minutes and 56 seconds
(11/15) Running minio-js tests ... done in 53 seconds
(12/15) Running minio-py tests ... FAILED in 5 minutes and 31 seconds
{
  "name": "minio-py:test_presigned_get_object_version",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-5b8780fa-9fcf-425e-bb02-2d681a29bace",
    "object_name": "9ad4c336-6491-4868-8971-19f491898000"
  },
  "message": "S3 operation failed; code: BucketAlreadyOwnedByYou, message: Your previous request to create the named bucket succeeded and you already own it., resource: /minio-py-test-5b8780fa-9fcf-425e-bb02-2d681a29bace, request_id: 165A4EFC2228945E, host_id: e43ba391-de86-4415-a490-a56d348f6171, bucket_name: minio-py-test-5b8780fa-9fcf-425e-bb02-2d681a29bace",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1448, in test_presigned_get_object_version\n    _CLIENT.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 624, in make_bucket\n    headers=headers,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 385, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketAlreadyOwnedByYou, message: Your previous request to create the named bucket succeeded and you already own it., resource: /minio-py-test-5b8780fa-9fcf-425e-bb02-2d681a29bace, request_id: 165A4EFC2228945E, host_id: e43ba391-de86-4415-a490-a56d348f6171, bucket_name: minio-py-test-5b8780fa-9fcf-425e-bb02-2d681a29bace\n",
  "duration": 120032
}
(12/15) Running s3cmd tests ... done in 33 seconds
(13/15) Running s3select tests ... FAILED in 2 minutes and 8 seconds
{
  "name": "s3select:test_csv_output_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-e361c8d7-6340-49d9-be0d-a74d861491c3"
  },
  "duration": 120075,
  "message": "S3 operation failed; code: BucketAlreadyOwnedByYou, message: Your previous request to create the named bucket succeeded and you already own it., resource: /s3select-test-e361c8d7-6340-49d9-be0d-a74d861491c3, request_id: 165A4F21884B052E, host_id: e43ba391-de86-4415-a490-a56d348f6171, bucket_name: s3select-test-e361c8d7-6340-49d9-be0d-a74d861491c3",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 57, in main\n    test_csv_output_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 148, in test_csv_output_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 624, in make_bucket\n    headers=headers,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 385, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketAlreadyOwnedByYou, message: Your previous request to create the named bucket succeeded and you already own it., resource: /s3select-test-e361c8d7-6340-49d9-be0d-a74d861491c3, request_id: 165A4F21884B052E, host_id: e43ba391-de86-4415-a490-a56d348f6171, bucket_name: s3select-test-e361c8d7-6340-49d9-be0d-a74d861491c3\n",
  "status": "FAIL"
}
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="43de5cc1-5f7e-4268-b4b9-a43147c64522"></a>
### 11280-dc0cf7e/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:31593
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 06f7b6f846e1:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 9 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 2 minutes and 26 seconds
(4/15) Running aws-sdk-ruby tests ... done in 21 seconds
(5/15) Running awscli tests ... done in 2 minutes and 49 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 4 minutes and 10 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 42 seconds
(9/15) Running minio-go tests ... done in 6 minutes and 38 seconds
(10/15) Running minio-java tests ... FAILED in 9 minutes and 2 seconds
{
  "name": "minio-java",
  "function": "putObject()",
  "args": "[user metadata]",
  "duration": 189,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = AuthenticationFailed, message = -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, github.com/Azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42\n===== RESPONSE ERROR (ServiceCode=AuthenticationFailed) =====\nDescription=Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.\nRequestId:aaeb44bb-c01e-00e6-4cfc-eafc1e000000\nTime:2021-01-15T05:09:32.0717349Z, Details: \n   AuthenticationErrorDetail: The MAC signature found in the HTTP request 'bqL4yE5wJ88ez3KYBnCOsid8QOhnJmszz9tTYUooUb8=' is not the same as any computed signature. Server used following string to sign: 'PUT\n\n\n128\n\napplication/xml\n\n\n\n\n\n\nx-ms-blob-cache-control:\nx-ms-blob-content-disposition:\nx-ms-blob-content-encoding:\nx-ms-blob-content-language:\nx-ms-blob-content-type:application/octet-stream\nx-ms-client-request-id:f93a90b9-3b7f-44b8-6f08-6e1af73415f0\nx-ms-date:Fri, 15 Jan 2021 05:09:32 GMT\nx-ms-meta-my_header1:a   b   c\nx-ms-meta-my_header2:\"a   b   c\"\nx-ms-meta-my_project:Project One\nx-ms-meta-my_unicode_tag:å•†å“\nx-ms-version:2019-02-02\n/minioazure/minio-java-test-1v1ets/minio-java-test-knriqp\ncomp:blocklist\ntimeout:1501'.\n   Code: AuthenticationFailed\n   PUT https://minioazure.blob.core.windows.net/minio-java-test-1v1ets/minio-java-test-knriqp?comp=blocklist&timeout=1501\n   Authorization: REDACTED\n   Content-Length: [128]\n   Content-Type: [application/xml]\n   User-Agent: [APN/1.0 MinIO/1.0 MinIO/2021-01-15T04:35:52Z]\n   X-Ms-Blob-Cache-Control: []\n   X-Ms-Blob-Content-Disposition: []\n   X-Ms-Blob-Content-Encoding: []\n   X-Ms-Blob-Content-Language: []\n   X-Ms-Blob-Content-Type: [application/octet-stream]\n   X-Ms-Client-Request-Id: [f93a90b9-3b7f-44b8-6f08-6e1af73415f0]\n   X-Ms-Date: [Fri, 15 Jan 2021 05:09:32 GMT]\n   X-Ms-Meta-My_header1: [a   b   c]\n   X-Ms-Meta-My_header2: [\"a   b   c\"]\n   X-Ms-Meta-My_project: [Project One]\n   X-Ms-Meta-My_unicode_tag: [商品]\n   X-Ms-Version: [2019-02-02]\n   --------------------------------------------------------------------------------\n   RESPONSE Status: 403 Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.\n   Content-Length: [1090]\n   Content-Type: [application/xml]\n   Date: [Fri, 15 Jan 2021 05:09:31 GMT]\n   Server: [Microsoft-HTTPAPI/2.0]\n   X-Ms-Error-Code: [AuthenticationFailed]\n   X-Ms-Request-Id: [aaeb44bb-c01e-00e6-4cfc-eafc1e000000]\n\n\n, bucketName = minio-java-test-1v1ets, objectName = minio-java-test-knriqp, resource = /minio-java-test-1v1ets/minio-java-test-knriqp, requestId = 165A4FA188B76066, hostId = 1284fd4e-4853-46fb-ab8f-1a44dc76c58b)\nrequest={method=PUT, url=http://minio-c2.minio.io:31593/minio-java-test-1v1ets/minio-java-test-knriqp, headers=x-amz-meta-My-Unicode-Tag: 商品\nx-amz-meta-My-Project: Project One\nx-amz-meta-My-header1: a   b   c\nx-amz-meta-My-Header2: \"a   b   c\"\nContent-Type: application/octet-stream\nHost: minio-c2.minio.io:31593\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: A9oFTxee7YVcJ9fWsgQeKg==\nx-amz-content-sha256: 1ff7959f86334ddc5c188a5083268f600146328b2b6c5185e75bf7d9387d6b74\nx-amz-date: 20210115T050931Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210115/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date;x-amz-meta-my-header1;x-amz-meta-my-header2;x-amz-meta-my-project;x-amz-meta-my-unicode-tag, Signature=*REDACTED*\n}\nresponse={code=403, headers=Accept-Ranges: bytes\nContent-Length: 3078\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 165A4FA188B76066\nX-Xss-Protection: 1; mode=block\nDate: Fri, 15 Jan 2021 05:09:32 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.putObject(MinioClient.java:4547), io.minio.MinioClient.putObject(MinioClient.java:2713), io.minio.MinioClient.putObject(MinioClient.java:2830), FunctionalTest.testPutObject(FunctionalTest.java:763), FunctionalTest.putObject(FunctionalTest.java:890), FunctionalTest.runObjectTests(FunctionalTest.java:3751), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(10/15) Running minio-js tests ... done in 2 minutes and 46 seconds
(11/15) Running minio-py tests ... done in 18 minutes and 23 seconds
(12/15) Running s3cmd tests ... done in 2 minutes and 20 seconds
(13/15) Running s3select tests ... done in 57 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="b38cf153-07f9-43cd-80e4-ff11a5e54bbd"></a>
### 11280-dc0cf7e/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:32581
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 34621b72a28b:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 45 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 2 minutes and 12 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... FAILED in 34 seconds
{
  "name": "mc",
  "duration": "9070",
  "function": "test_mirror_list_objects_storage_class",
  "status": "FAIL",
  "error": "mirror and list differs"
}
(7/15) Running minio-dotnet tests ... done in 52 seconds
(8/15) Running minio-go tests ... done in 2 minutes and 27 seconds
(9/15) Running minio-java tests ... done in 1 minutes and 54 seconds
(10/15) Running minio-js tests ... done in 51 seconds
(11/15) Running minio-py tests ... done in 6 minutes and 12 seconds
(12/15) Running s3cmd tests ... done in 33 seconds
(13/15) Running s3select tests ... done in 1 minutes and 8 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/erasure-multipart.go
+++ b/cmd/erasure-multipart.go
@@ -690,7 +690,7 @@ func (er erasureObjects) CompleteMultipartUpload(ctx context.Context, bucket str
 
 	// Check if an object is present as one of the parent dir.
 	// -- FIXME. (needs a new kind of lock).
-	if er.parentDirIsObject(ctx, bucket, path.Dir(object)) {
+	if opts.ParentIsObject != nil && opts.ParentIsObject(ctx, bucket, path.Dir(object)) {
 		return oi, toObjectErr(errFileParentIsFile, bucket, object)
 	}
 

--- a/cmd/erasure-object.go
+++ b/cmd/erasure-object.go
@@ -621,7 +621,7 @@ func (er erasureObjects) putObject(ctx context.Context, bucket string, object st
 	// Check if an object is present as one of the parent dir.
 	// -- FIXME. (needs a new kind of lock).
 	// -- FIXME (this also causes performance issue when disks are down).
-	if er.parentDirIsObject(ctx, bucket, path.Dir(object)) {
+	if opts.ParentIsObject != nil && opts.ParentIsObject(ctx, bucket, path.Dir(object)) {
 		return ObjectInfo{}, toObjectErr(errFileParentIsFile, bucket, object)
 	}
 

--- a/cmd/erasure-sets.go
+++ b/cmd/erasure-sets.go
@@ -773,8 +773,13 @@ func (s *erasureSets) GetObject(ctx context.Context, bucket, object string, star
 	return s.getHashedSet(object).GetObject(ctx, bucket, object, startOffset, length, writer, etag, opts)
 }
 
+func (s *erasureSets) parentDirIsObject(ctx context.Context, bucket, parent string) bool {
+	return s.getHashedSet(parent).parentDirIsObject(ctx, bucket, parent)
+}
+
 // PutObject - writes an object to hashedSet based on the object name.
 func (s *erasureSets) PutObject(ctx context.Context, bucket string, object string, data *PutObjReader, opts ObjectOptions) (objInfo ObjectInfo, err error) {
+	opts.ParentIsObject = s.parentDirIsObject
 	return s.getHashedSet(object).PutObject(ctx, bucket, object, data, opts)
 }
 
@@ -1064,6 +1069,7 @@ func (s *erasureSets) AbortMultipartUpload(ctx context.Context, bucket, object, 
 
 // CompleteMultipartUpload - completes a pending multipart transaction, on hashedSet based on object name.
 func (s *erasureSets) CompleteMultipartUpload(ctx context.Context, bucket, object, uploadID string, uploadedParts []CompletePart, opts ObjectOptions) (objInfo ObjectInfo, err error) {
+	opts.ParentIsObject = s.parentDirIsObject
 	return s.getHashedSet(object).CompleteMultipartUpload(ctx, bucket, object, uploadID, uploadedParts, opts)
 }
 

--- a/cmd/metacache-set.go
+++ b/cmd/metacache-set.go
@@ -680,8 +680,9 @@ func (er *erasureObjects) listPath(ctx context.Context, o listPathOptions) (entr
 				logger.LogIf(ctx, err)
 				custom := b.headerKV()
 				_, err = er.putObject(ctx, minioMetaBucket, o.objectPath(b.n), NewPutObjReader(r, nil, nil), ObjectOptions{
-					UserDefined: custom,
-					NoLock:      true, // No need to hold namespace lock, each prefix caches uniquely.
+					UserDefined:    custom,
+					NoLock:         true, // No need to hold namespace lock, each prefix caches uniquely.
+					ParentIsObject: nil,
 				})
 				if err != nil {
 					metaMu.Lock()

--- a/cmd/object-api-interface.go
+++ b/cmd/object-api-interface.go
@@ -44,16 +44,16 @@ type ObjectOptions struct {
 	MTime                time.Time // Is only set in POST/PUT operations
 	Expires              time.Time // Is only used in POST/PUT operations
 
-	DeleteMarker                  bool                   // Is only set in DELETE operations for delete marker replication
-	UserDefined                   map[string]string      // only set in case of POST/PUT operations
-	PartNumber                    int                    // only useful in case of GetObject/HeadObject
-	CheckPrecondFn                CheckPreconditionFn    // only set during GetObject/HeadObject/CopyObjectPart preconditional valuation
-	DeleteMarkerReplicationStatus string                 // Is only set in DELETE operations
-	VersionPurgeStatus            VersionPurgeStatusType // Is only set in DELETE operations for delete marker version to be permanently deleted.
-	TransitionStatus              string                 // status of the transition
-	NoLock                        bool                   // indicates to lower layers if the caller is expecting to hold locks.
-	ProxyRequest                  bool                   // only set for GET/HEAD in active-active replication scenario
-
+	DeleteMarker                  bool                                                  // Is only set in DELETE operations for delete marker replication
+	UserDefined                   map[string]string                                     // only set in case of POST/PUT operations
+	PartNumber                    int                                                   // only useful in case of GetObject/HeadObject
+	CheckPrecondFn                CheckPreconditionFn                                   // only set during GetObject/HeadObject/CopyObjectPart preconditional valuation
+	DeleteMarkerReplicationStatus string                                                // Is only set in DELETE operations
+	VersionPurgeStatus            VersionPurgeStatusType                                // Is only set in DELETE operations for delete marker version to be permanently deleted.
+	TransitionStatus              string                                                // status of the transition
+	NoLock                        bool                                                  // indicates to lower layers if the caller is expecting to hold locks.
+	ProxyRequest                  bool                                                  // only set for GET/HEAD in active-active replication scenario
+	ParentIsObject                func(ctx context.Context, bucket, parent string) bool // Used to verify if parent is an object.
 }
 
 // BucketOptions represents bucket options for ObjectLayer bucket operations

